core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -500,6 +501,70 @@ def validate_postcodes(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRoun
         ]
 
 
+def validate_current_project_delivery_status_and_upcoming_comms_milestones(
+    workbook: dict[str, pd.DataFrame]
+) -> list["TownsFundRoundFourValidationFailure"] | None:
+    """
+    Validates the Project Progress table in the Round 4 submission workbook.
+
+    Checks for rows with a Project Delivery Status other than "4. Completed" where both
+    the Current Project Delivery Stage and Most Important Upcoming Comms Milestone are null.
+    If such invalid states are found, returns a list of TownsFundRoundFourValidationFailure instances.
+
+    :param workbook: A dictionary containing sheet names as keys and pandas DataFrames
+                     representing each sheet in the Round 4 submission.
+    :return: A list of TownsFundRoundFourValidationFailure instances if validation failures occur,
+             otherwise returns None.
+    """
+    project_progress_df = workbook["Project Progress"]
+
+    project_incomplete_rows = project_progress_df[project_progress_df["Project Delivery Status"] != "4. Completed"]
+
+    na_values = {"", "< Select >", np.nan, None}
+
+    columns_to_check = [
+        "Current Project Delivery Stage",
+        "Most Important Upcoming Comms Milestone",
+        "Date of Most Important Upcoming Comms Milestone (e.g. Dec-22)",
+    ]
+
+    invalid_rows = pd.DataFrame()
+
+    for column in columns_to_check:
+        invalid_rows = pd.concat([invalid_rows, find_invalid_rows(project_incomplete_rows, column, na_values)], axis=0)
```

Comment:
This is quite hard to follow and is a new way of doing this check. Can it be simplified?